### PR TITLE
Library information scraper #31

### PR DESCRIPTION
**Pull request by g3wanghc**
Includes fixed for scraper.get() 404 limbo #62

### CODE REVIEW

File: uoftscrapers/scrapers/libraries/__init__.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+    @staticmethod
+    def scrape(location='.'):
+        Scraper.logger.info('Libraries initialized.')
+        Scraper.ensure_location(location)
+
+        for library_link in Libraries.get_library_link():
+            doc = Libraries.get_library_doc(library_link)
+            if (doc is not None):
+                Scraper.save_json(doc, location, library_link.split('/')[-1])
```
Comment (kashav) [inline uoftscrapers/scrapers/libraries/__init__.py]: I think all filenames should be capitalized, also might be a good idea to add an `id` key with this same value for consistency.

File: uoftscrapers/scrapers/libraries/__init__.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+        doc = OrderedDict([
+            ('name', library_name),
+            ('image', library_image),
+            ('website', library_website),
+            ('hours', library_hours),
+            ('address', library_address),
+            ('phone', library_phone),
+            ('about', library_about),
+            ('collection_strenghts', library_collection_strenghts),
```
Comment (kashav) [inline uoftscrapers/scrapers/libraries/__init__.py]: "strengths" is misspelled

File: uoftscrapers/scrapers/libraries/__init__.py (hunk 1 above)
Comment (g3wanghc) [inline uoftscrapers/scrapers/libraries/__init__.py]: @kshvmdn The names should be unique, I will capitalize them.

**Comment by kashav** [comment]
For `hours`, I think it'd be better if each day was a separate object with a `closed` boolean and `open` / `closed` number, something like:

``` js
"hours": {
  "sunday": {
    "closed": Boolean,
    "open": Number,
    "closed": Number
  },
  "monday": {...},
  "tuesday": {...},
  ...
}
```

We do this for the [food scraper](https://github.com/cobalt-uoft/uoft-scrapers/blob/master/uoftscrapers/scrapers/food/__init__.py#L77) and it makes it a lot easier to filter data.

Other than that, I think it looks good!

File: uoftscrapers/scrapers/utils/scraper.py [hunk 3]
```diff
@@ -39,9 +39,12 @@ def get(url, params=None, cookies=None, headers=None, json=False, max_attempts=1
...
                     doc = r
                 else:
+                    attempts += 1
                     sleep(0.5)
             except (requests.exceptions.Timeout,
-                    requests.exceptions.ConnectionError):
+                    requests.exceptions.ConnectionError,
+                    requests.exceptions.InvalidSchema,
+                    requests.exceptions.MissingSchema):
                 attempts += 1
```
Comment (qasim) [inline uoftscrapers/scrapers/utils/scraper.py:48]: 👌

**Comment by g3wanghc** [comment]
@kshvmdn @qasim is the use of decimals a common thing to denote time?

**Comment by qasim** [comment]
@g3wanghc storing time (just time, no date information) is generally done by giving a number which represents a unit of time since midnight (in this case, the number is the hours since midnight). We can move this to seconds which would avoid the need to use decimals, but it hasn't been hurting us so far.

**Comment by g3wanghc** [comment]
@qasim Can I keep the hours as a numeric string in order to keep leading/trailing zeros. (e.g. '09:30') 
Personally I think storing time in terms of decimals is pretty sketch in the long run.

**Comment by g3wanghc** [comment]
If this is the standard time-notation, I can convert the start-time and end-time for the Events scrapper as well.

**Comment by qasim** [comment]
I don't know whether we can move to time being a string, mainly because of how we use the number format. Eventually, when APIs are made for the scrapers, it let's us do numerical queries on time without that much friction (in the filter endpoints). For example, you can do `start:>10` to represent "start time after 10AM".

However, I do agree that the decimals are weird. Maybe this is a good time to move to "seconds since midnight" instead of "hours since midnight" as to avoid decimals entirely.

Opinions? @arkon @kshvmdn

**Comment by qasim** [comment]
I'm also open to there being 2 time keys, sort of like `"time":Number` and `"time_str":String`.

**Comment by g3wanghc** [comment]
I'm would definitely prefer using seconds to midnight consistently. The risk of rounding errors on `"time":Number` is a bit too weird for me.

**Comment by kashav** [comment]
I think using hours is cleaner just for the reason that they're more natural to work with (`time:10.75` vs `time:38700`), but I also agree that decimal hours can get messy, esp. with _odd_ times (eg. 10:37 AM, 4:28 PM, etc).

It's probably better to make the switch now, since we'll probably need to work with times like these sometime in the future.

So I guess +1 for seconds.

**Comment by qasim** [comment]
@kshvmdn I think in the future we can work on something like `time:>"10:45"` and then our query parser would handle converting it to seconds to put into the filter.

**Comment by g3wanghc** [comment]
@qasim @kshvmdn done :D

File: uoftscrapers/scrapers/libraries/__init__.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+        doc = OrderedDict([
+            ('name', library_name),
+            ('image', library_image),
+            ('website', library_website),
+            ('hours', library_hours),
+            ('address', library_address),
+            ('phone', library_phone),
+            ('about', library_about),
+            ('collection_strenghts', library_collection_strengths),
```
Comment (kashav) [inline uoftscrapers/scrapers/libraries/__init__.py]: `collection_strenghts` (key) is still misspelled

**Comment by kashav** [comment]
Other than that misspelling, I think it looks good!

Also, do you think it'd be better to use `null` instead of `0` for hours when the place is closed?

**Comment by g3wanghc** [comment]
@kshvmdn Depends on the type handling for the end user. Personally I prefer having `null` but I think it is simpler to keep it as a `Number`. Either way, those values shouldn't be trusted if the day is marked as `closed`. ¯_(ツ)_/¯

**Comment by qasim** [comment]
LGTM

Thanks for the bug fix + a shiny new scraper!

**Pull request merged**